packages/gatsby/src/utils/webpack.config.js

```diff
@@ -582,12 +588,7 @@ module.exports = async (
           framework: {
             chunks: `all`,
             name: `framework`,
-            // This regex ignores nested copies of framework libraries so they're bundled with their issuer.
-            test: new RegExp(
-              `(?<!node_modules.*)[\\\\/]node_modules[\\\\/](${FRAMEWORK_BUNDLES.join(
-                `|`
-              )})[\\\\/]`
-            ),
+            test: FRAMEWORK_BUNDLES_REGEX,
```

Comment:
In develop stage we don't need to change anything, `react-dom/server` is only relevant for `build-javascript` stage